pennylane_lightning/core/src/observables/Observables.hpp

```diff
@@ -414,4 +414,114 @@ class HamiltonianBase : public Observable<StateVectorT> {
     }
 };
 
+/**
+ * @brief Sparse representation of SparseHamiltonian<T>
+ *
+ * @tparam T Floating-point precision.
+ */
+template <class StateVectorT>
+class SparseHamiltonianBase : public Observable<StateVectorT> {
+  public:
+    using PrecisionT = typename StateVectorT::PrecisionT;
+    using ComplexT = typename StateVectorT::ComplexT;
+#ifdef _ENABLE_PLGPU
+    using IdxT =
+        typename std::conditional<std::is_same<PrecisionT, float>::value,
+                                  int32_t, int64_t>::type;
+#else
+    using IdxT = std::size_t;
+#endif
+
+  protected:
+    std::vector<ComplexT> data_;
+    std::vector<IdxT> indices_;
+    std::vector<IdxT> offsets_;
+    std::vector<std::size_t> wires_;
+
+  private:
+    // LCOV_EXCL_START
+    [[nodiscard]] bool
+    isEqual(const Observable<StateVectorT> &other) const override {
+        const auto &other_cast =
+            static_cast<const SparseHamiltonianBase<StateVectorT> &>(other);
+        return data_ != other_cast.data_ || indices_ != other_cast.indices_ ||
+               offsets_ != other_cast.offsets_;
+    }
+    // LCOV_EXCL_STOP
```

Comment:
Why are we excluding it from coverage?

Reply:
Included following discussion with @multiphaseCFD .